### DISCUSSION TITLE
Switching to content-driven framework

### DISCUSSION THREAD

**Discussion by bartosz-skejcik** [Buzzpy/Dev-Encyclopedia#27] [Ideas]
As a front-end developer, I'm wandering. Why don't we switch to, for example Astro.js? It's a framework built exactly for the type of websites that Dev Encyclopedia is.

**Comment by bartosz-skejcik** [comment]
I'd be more than happy to make the first move in switching to Astro.js. I could write all the content that there is now and leave a detailed documentation for adding new things

**Reply by ravi-teja-nani** [reply]
Agree with you, i also thought the same. Could be better to move next js or astro

**Reply by bartosz-skejcik** [reply]
I think there is no need to move to next.js, as it's a framework for highly sophisticated and complex apps. Here we are building a simple web app and all the data is already pre-rendered because it's static.

**Reply by bartosz-skejcik** [reply]
@Buzzpy what’s your take on that?

**Reply by Buzzpy** [reply]
I think that'd be awesome. Please post your opinion to my latest comment on issue #38 so that can all come to a conclusion there!

Thanks again for your efforts. 

![Copy of assets (3)](https://github.com/user-attachments/assets/69e63840-99f9-4f62-8630-d740b0ab8846)

**Comment by bartosz-skejcik** [comment]
Here is a link for the astro website where you can read more about it. https://astro.build/

**Comment by bartosz-skejcik** [comment]
I made the switch already ;) [here](https://github.com/bartosz-skejcik/Dev-Encyclopedia) is the converted site. tell me if you want to go through with it :)

The CI/CD will work with astro cause when building the project it outputs clean html, css and js :smiley: 

We can also improve the functionality further cause we can add a react component for the modal so it's easier to manage cause for now im using the data.json that @4rnv provided in his [idea](https://github.com/Buzzpy/Dev-Encyclopedia/discussions/29)

**Reply by Buzzpy** [reply]
It's done! We are deploying the Astro-integrated site tomorrow!
